lib/vector/Vlib/geos.c

```diff
@@ -362,15 +362,24 @@ GEOSCoordSequence *V1_read_line_geos(struct Map_info *Map, long offset,
     else
         z = NULL;
 
-    if (0 >= dig__fread_port_D(x, n_points, &(Map->dig_fp)))
-        return NULL; /* end of file */
+    if (0 >= dig__fread_port_D(x, n_points, &(Map->dig_fp))) {
+        GEOSCoordSeq_destroy(pseq);
```

Comment:
If you move the `pseq = GEOSCoordSeq_create…`to just before `pseq` is used, then no need to destroy and set to NULL. Just need to initialise to NULL.

Reply:
done

Reply:
> Just need to initialise to NULL.

Reply:
done